javascript/ql/src/Security/CWE-116/IncompleteSanitization.ql

```diff
@@ -20,53 +20,47 @@ import javascript
 string metachar() { result = "'\"\\&<>\n\r\t*|{}[]%$".charAt(_) }
 
 /** Gets a string matched by `e` in a `replace` call. */
-string getAMatchedString(Expr e) {
-  result = getAMatchedConstant(e.(RegExpLiteral).getRoot()).getValue()
+string getAMatchedString(DataFlow::Node e) {
+  result = e.(DataFlow::RegExpLiteralNode).getRoot().getAMatchedString()
```

Comment:
Can we afford to generalize to support data flow and constructed regular expressions?

I.e. reformualte this disjunct to something like:

```
exists (RegExpPatternSource src | 
   src.getARegExpObject().flowsTo(e) |
   src ... // make use of the newly introduced strings-as-parsed-regexps
)
```

Reply:
Yes, I think that should be possible. (The new regexp parsing hadn't been merged yet when I started working on this.)

Reply:
It's actually a little more involved than I thought: this query wants to reason about whether a regular expression is global. `RegExpPatternSource` is not the right API to answer that question, so we need a few more bits to make it work. Nothing very complicated, but more than just a few lines. Would you like me to add it to this PR or defer?